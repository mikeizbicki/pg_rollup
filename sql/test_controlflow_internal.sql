SET client_min_messages TO WARNING;

create temporary table testinternal (
    id serial primary key,
    name text,
    num int
);


select create_rollup_internal(
    'testinternal_internal1',
    joininfos => '[{"table_name":"testinternal","table_alias":"testinternal","condition":"","join_type":"FROM"}]',
    columns => '{{avg(num),avg}}',
    groups => '{{name,name},{num,num}}',
    having_clause => 'num = 1 OR num >3',
    dry_run => False
);

select create_rollup_internal(
    'testinternal_internal2',
    joininfos => '[{"table_name":"testinternal","table_alias":"testinternal","condition":"","join_type":"FROM"}]',
    columns => '{{sum(num),sum},{max(num),max}}',
    groups => '{{name,name}}',
    where_clause => 'num = 1 OR num >3',
    dry_run => False
);

insert into testinternal (name,num) values
    ('alice', 1),
    ('alice', 2),
    ('alice', 3),
    ('alice', 4),
    ('alice', 5),
    ('bill', 5),
    ('bill', 5),
    ('bill', 5),
    ('charlie', 1),
    ('charlie', 1),
    ('charlie', 1),
    ('charlie', 3),
    ('charlie', NULL),
    ('dave', 4),
    ('elliot', 5),
    (NULL, 1),
    (NULL, 1),
    (NULL, 1),
    (NULL, 1),
    (NULL, 1),
    (NULL, 8),
    (NULL, 9),
    (NULL, NULL),
    (NULL, NULL),
    (NULL, NULL),
    (NULL, NULL);

select * from testinternal_internal1;
select assert_rollup('testinternal_internal1');
select assert_rollup('testinternal_internal2');

drop table testinternal cascade;
