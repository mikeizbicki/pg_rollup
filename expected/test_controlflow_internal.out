SET client_min_messages TO WARNING;
create or replace language plpython3u;
create extension if not exists pg_rollup;
create temporary table testinternal (
    id serial primary key,
    name text,
    num int
);
select create_rollup_internal(
    'testinternal',
    'testinternal_internal1',
    columns => '{{avg(num),avg}}',
    groups => '{{name,name},{num,num}}',
    having_clause => 'num = 1 OR num >3',
    dry_run => False
);
 create_rollup_internal 
------------------------
 
(1 row)

select create_rollup_internal(
    'testinternal',
    'testinternal_internal2',
    columns => '{{sum(num),sum},{max(num),max}}',
    groups => '{{name,name}}',
    where_clause => 'num = 1 OR num >3',
    dry_run => False
);
 create_rollup_internal 
------------------------
 
(1 row)

insert into testinternal (name,num) values
    ('alice', 1),
    ('alice', 2),
    ('alice', 3),
    ('alice', 4),
    ('alice', 5),
    ('bill', 5),
    ('bill', 5),
    ('bill', 5),
    ('charlie', 1),
    ('charlie', 1),
    ('charlie', 1),
    ('charlie', 3),
    ('charlie', NULL),
    ('dave', 4),
    ('elliot', 5),
    (NULL, 1),
    (NULL, 1),
    (NULL, 1),
    (NULL, 1),
    (NULL, 1),
    (NULL, 8),
    (NULL, 9),
    (NULL, NULL),
    (NULL, NULL),
    (NULL, NULL),
    (NULL, NULL);
select * from testinternal_internal1;
 avg | group_name | group_num 
-----+------------+-----------
   8 |            |         8
   9 |            |         9
   1 |            |         1
   1 | alice      |         1
   5 | bill       |         5
   5 | elliot     |         5
   5 | alice      |         5
   1 | charlie    |         1
   4 | dave       |         4
   4 | alice      |         4
(10 rows)

select assert_rollup('testinternal_internal1');
 assert_rollup 
---------------
 
(1 row)

select assert_rollup('testinternal_internal2');
 assert_rollup 
---------------
 
(1 row)

